runtime/vm/jvminit.c

```diff
@@ -6323,7 +6325,8 @@ predefinedHandlerWrapper(struct J9PortLibrary *portLibrary, U_32 gpType, void *g
 }
 
 IDATA
-registerPredefinedHandler(J9JavaVM *vm, U_32 signal, void **oldOSHandler) {
+registerPredefinedHandler(J9JavaVM *vm, U_32 signal, void **oldOSHandler)
```

Comment:
Do we need two copies of this code?  Could this one call `registerOSHandler` with `predefinedHandlerWrapper` as the `newOSHandler`?  Or can we push knowledge of which handler to pass to the caller?

Reply:
`registerPredefinedHandler` sets OS handler as `omrsignal.c::masterASynchSignalHandler` [not exposed externally], and adds `predefinedHandlerWrapper` to the `asyncHandlerList` list. `predefinedHandlerWrapper` is not an OS handler.

`registerOSHandler` registers the user-provided OS handler with specified signal.

we can merge `registerOSHandler` and `registerPredefinedHandler`. use `newOSHandler` as `J9_PRE_DEFINED_HANDLER_CHECK(2)` to specify the `predefinedHandlerWrapper`. should i proceed with this?

Reply:
You're right.  The two functions are subtly different.  Better to leave them as two different calls.